pkg/template/kurl_context.go

```diff
@@ -0,0 +1,197 @@
+	}
+
+	return ctx
+}
+
+func (ctx KurlCtx) AddValuesToKurlContext(retrieved *kurlv1beta1.Installer) {
+	Spec := reflect.ValueOf(retrieved.Spec)
+
+	for i := 0; i < Spec.NumField(); i++ {
+		Category := reflect.ValueOf(Spec.Field(i).Interface())
+
+		TypeOfCategory := Category.Type()
+
+		RawCategoryName := Category.String()
+		TrimmedRight := strings.Split(RawCategoryName, ".")[1]
+		CategoryName := strings.Split(TrimmedRight, " ")[0]
+
+		for i := 0; i < Category.NumField(); i++ {
+			if Category.Field(i).CanInterface() {
+				ctx.KurlValues[CategoryName+"."+TypeOfCategory.Field(i).Name] = Category.Field(i).Interface()
+			}
+		}
+	}
+}
+
+type KurlCtx struct {
+	KurlValues map[string]interface{}
+}
+
+func (ctx KurlCtx) FuncMap() template.FuncMap {
+	return template.FuncMap{
+		"KurlString": ctx.kurlString,
+		"KurlInt":    ctx.kurlInt,
+		"KurlBool":   ctx.kurlBool,
+		"KurlOption": ctx.kurlOption,
+		"KurlAll":    ctx.kurlAll,
+	}
+}
+
+func (ctx KurlCtx) kurlBool(yamlPath string) bool {
+	if len(ctx.KurlValues) == 0 {
+		fmt.Printf("No Installer CRD has been found, is this a kURL cluster?\n")
+		return false
+	}
+
+	result, ok := ctx.KurlValues[yamlPath]
+	if !ok {
+		fmt.Printf("There is no value found at the yamlPath ''%s'\n", yamlPath)
+		return false
+	}
+
+	b, ok := result.(bool)
+	if !ok {
+		fmt.Printf("The yamlPath '%s' corresponds to value '%v' of type '%T'. The KurlBool function supports only boolean values\n", yamlPath, result, result)
+		return false
+	}
+
+	return b
+}
+
+func (ctx KurlCtx) kurlInt(yamlPath string) int {
+	if len(ctx.KurlValues) == 0 {
+		fmt.Printf("No Installer CRD has been found, is this a kURL cluster?\n")
+		return 0
+	}
+
+	result, ok := ctx.KurlValues[yamlPath]
+	if !ok {
+		fmt.Printf("There is no value found at the yamlPath '%s'\n", yamlPath)
+		return 0
+	}
+
+	i, ok := result.(int)
+	if !ok {
+		fmt.Printf("The yamlPath '%s' corresponds to value '%v' of type '%T'. The KurlInt function supports only integer values\n", yamlPath, result, result)
+		return 0
+	}
+
+	return i
+}
+
+func (ctx KurlCtx) kurlString(yamlPath string) string {
+	if len(ctx.KurlValues) == 0 {
+		fmt.Printf("No Installer CRD has been found, is this a kURL cluster?\n")
+		return ""
+	}
+
+	result, ok := ctx.KurlValues[yamlPath]
+	if !ok {
+		fmt.Printf("There is no value found at the yamlPath '%s'\n", yamlPath)
+		return ""
+	}
+
+	s, ok := result.(string)
+	if !ok {
+		fmt.Printf("The yamlPath '%s' corresponds to value '%v' of type '%T'. The KurlString function supports only string values\n", yamlPath, result, result)
+		return ""
+	}
+
+	return s
+}
+
+func (ctx KurlCtx) kurlOption(yamlPath string) string {
+	if len(ctx.KurlValues) == 0 {
+		fmt.Printf("No Installer CRD has been found, is this a kURL cluster?\n")
+		return ""
+	}
+
+	result, ok := ctx.KurlValues[yamlPath]
+	if !ok {
+		fmt.Printf("There is no value found at the yamlPath '%s'\n", yamlPath)
+		return ""
+	}
+
+	switch t := interface{}(result).(type) {
+	case int:
```

Comment:
don't really need this.   this works for anything that can be "stringified": `fmt.Sprintf("%v", strOrBoolOrInt)`